### PR TITLE
Implement exercise spiral-matrix

### PR DESCRIPTION
**Pull request by MattH-be**

### CODE REVIEW

File: exercises/spiral-matrix/spiral-matrix.spec.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import SpiralMatrix from './spiral-matrix';
```
Comment (rchavarria) [inline exercises/spiral-matrix/spiral-matrix.spec.js]: Very clear formatting. All different matrix are clearly readable.

File: exercises/spiral-matrix/example.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class SpiralMatrix {
+
+  static ofSize(size) {
+    const spiral = [];
+    for (; spiral.length < size; spiral.push([]));
```
Comment (rchavarria) [inline exercises/spiral-matrix/example.js]: Take a look at [`Array.prototype.fill`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill). I think there should be a more clear way of creating a matrix, kind of `new Array(n).fill(new Array(m)` or so.

File: exercises/spiral-matrix/example.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    const totalNumbers = size * size;
```
Comment (matthewmorgan) [inline exercises/spiral-matrix/example.js]: If you like, you can use the new exponentiation operator:

```javascript
const totalNumbers = size**2;
``` 

Node 7 should support it.

**Comment by MattH-be** [comment]
Just pushed the small changes you guys where asking for.

**Pull request merged**